improver/blending/blend_across_adjacent_points.py

```diff
@@ -155,11 +156,22 @@ def process(self, cube):
 
         # Calculate weights and produce blended output.
         weights = self.WeightsPlugin(cube, self.coord, self.central_point)
-        blended_cube = self.BlendingPlugin(cube, weights)
 
-        # With one threshold dimension (such as for low cloud), the threshold
-        # axis is demoted to a scalar co-ordinate by BlendingPlugin. This line
-        # promotes threshold to match the dimensions of central_point_cube.
-        blended_cube = check_cube_coordinates(central_point_cube, blended_cube)
+        dims_to_collapse = set(range(cube.ndim)) - {cube.coord_dims(self.coord)[0]}
+
+        # In the event of multiple dimensions, always use the first available one to slice over
+        dim_to_collapse = min(dims_to_collapse)
+
+        allow_slicing = not self.BlendingPlugin.check_percentile_coord(cube)
+
+        cube_slices = cube.slices_over(dim_to_collapse) if allow_slicing else [cube]
+        result_slices = CubeList()
+        for cube_slice in cube_slices:
+            result_slice = self.BlendingPlugin(cube_slice, weights=weights,)
+            result_slices.append(result_slice)
+        blended_cube = result_slices.merge_cube() if allow_slicing else result_slices[0]
+
+        # Copy the metadata of the central cube
         blended_cube = central_point_cube.copy(blended_cube.data)
+
```

Comment:
Sorry, I've only now realised that the common part is `WeightedBlendAcrossWholeDimension` plugin, so I suggest we make the modifications there instead:
```diff
diff --git a/improver/blending/weighted_blend.py b/improver/blending/weighted_blend.py
index 2e592e80..1d99d1ba 100644
--- a/improver/blending/weighted_blend.py
+++ b/improver/blending/weighted_blend.py
@@ -575,12 +575,12 @@ class WeightedBlendAcrossWholeDimension(PostProcessingPlugin):
             weights_array = self.shape_weights(cube, weights)
         else:
             (number_of_fields,) = cube.coord(self.blend_coord).shape
-            weights_array = np.broadcast_to(1.0 / number_of_fields, cube.shape).astype(
-                np.float32
+            weights_array = np.broadcast_to(
+                np.asarray(1.0 / number_of_fields, dtype=np.float32), cube.shape
             )
         (blend_dim,) = cube.coord_dims(self.blend_coord)
         self.check_weights(weights_array, blend_dim)
-        return weights_array.astype(np.float32)
+        return weights_array
 
     def percentile_weights(self, cube, weights, perc_coord):
         """
@@ -632,8 +632,8 @@ class WeightedBlendAcrossWholeDimension(PostProcessingPlugin):
             )
         else:
             (number_of_fields,) = cube.coord(self.blend_coord).shape
-            weights_array = np.broadcast_to(1.0 / number_of_fields, cube.shape).astype(
-                np.float32
+            weights_array = np.broadcast_to(
+                np.asarray(1.0 / number_of_fields, dtype=np.float32), cube.shape
             )
 
         (blend_dim,) = cube.coord_dims(self.blend_coord)
@@ -647,7 +647,7 @@ class WeightedBlendAcrossWholeDimension(PostProcessingPlugin):
         # Check the weights add up to 1 across the blending dimension.
         self.check_weights(weights_array, 0)
 
-        return weights_array.astype(np.float32)
+        return weights_array
 
     def percentile_weighted_mean(self, cube, weights, perc_coord):
         """
@@ -721,13 +721,23 @@ class WeightedBlendAcrossWholeDimension(PostProcessingPlugin):
         """
         weights_array = self.non_percentile_weights(cube, weights)
 
-        # Calculate the weighted average.
-        cube_new = collapsed(
-            cube, self.blend_coord, iris.analysis.MEAN, weights=weights_array
+        (collapse_dim,) = cube.coord_dims(self.blend_coord)
+        slice_dim = 1 if collapse_dim == 0 else 0
+        allow_slicing = cube.ndim > 1 and cube.shape[slice_dim] > 1
+
+        cube_slices = cube.slices_over(slice_dim) if allow_slicing else [cube]
+        weights_slices = (
+            np.moveaxis(weights_array, slice_dim, 0)
+            if allow_slicing
+            else [weights_array]
         )
-        cube_new.data = cube_new.data.astype(np.float32)
+        result_slices = iris.cube.CubeList(
+            collapsed(c_slice, self.blend_coord, iris.analysis.MEAN, weights=w_slice)
+            for c_slice, w_slice in zip(cube_slices, weights_slices)
+        )
+        result = result_slices.merge_cube() if allow_slicing else result_slices[0]
 
-        return cube_new
+        return result
 
     @staticmethod
     def _get_cycletime_point(input_cube, cycletime):
```